extensions/modules/src/org/exist/xquery/modules/compression/AbstractCompressFunction.java

```diff
@@ -432,7 +432,7 @@ private void compressResource(OutputStream os, DocumentImpl doc, boolean useHier
 		} else if (doc.getResourceType() == DocumentImpl.BINARY_FILE) {
 			// binary file
             try (final InputStream is = context.getBroker().getBinaryResource((BinaryDocument)doc);
-                 final FastByteArrayOutputStream baos = new FastByteArrayOutputStream((int)doc.getContentLength())) {
+                 final FastByteArrayOutputStream baos = new FastByteArrayOutputStream()) {
```

Comment:
Would you consider `doc.getContentLength() == -1 ? 1024 : (int)doc.getContentLength()` instead please